### PR TITLE
Imposed constraint update

### PR DESCRIPTION
**Pull request by djc**
Happened to find this change somewhat randomly:

https://hg.mozilla.org/projects/nss/rev/023f640c52040ff02bdd3c44df2b5ae927fa9e4f
https://groups.google.com/a/mozilla.org/g/dev-security-policy/c/DotaWOS0v1E/m/icFj9tAzBQAJ
https://bugzilla.mozilla.org/show_bug.cgi?id=1802944

Propose to publish yet another release after this is merged.

### CODE REVIEW

**Review by cpu — APPROVED** [review]
Nice spot

**Comment by djc** [comment]
I was looking at that earlier. Using that would enable noticing changes, but going from `*.tr` to the constraints encoding would be a manual step. For now, I've subscribed to https://hg.mozilla.org/projects/nss/atom-log/tip/lib/certdb/genname.c in my RSS reader.

**Comment by cpu** [comment]
> but going from *.tr to the constraints encoding would be a manual step. 

Why is that?

**Comment by ctz** [comment]
> Why is that?

I guess we'd need to write an encoder to put these strings into the form of a `NameConstraints` extension body, but that is definitely doable?

**Comment by ctz** [comment]
Way back I had added a test for this root's constraints -- unfortunately I never landed it because the test failed, because the webpki crate didn't have working name constraints support back then. That is fixed since January. The old commit is here: https://github.com/rustls/webpki-roots/commit/65282d9f07f46d9f272c82034811f9155920844b - unfortunately only a positive test.

File: tests/codegen.rs [hunk 1]
```diff
@@ -249,28 +249,7 @@ const TUBITAK1_SUBJECT_DN: &[&[u8]] = &[
...
-    b"\x30\x09\x82\x07",
-    b".kep.tr",
-    b"\x30\x09\x82\x07",
-    b".bel.tr",
-    b"\x30\x09\x82\x07",
-    b".edu.tr",
-    b"\x30\x09\x82\x07",
-    b".org.tr",
-];
+const TUBITAK1_NAME_CONSTRAINTS: &[&[u8]] = &[b"\x30\x09\xa0\x07", b"\x30\x05\x82\x03", b".tr"];
```
Comment (ctz) [inline tests/codegen.rs:252]: I think this wants 0xa0, 0x0b on  the front.

- 0xa0  = `CONTEXT_SPECIFIC 0 | CONSTRUCTED`)
- 0x0b  = 11 bytes length

**Review by ctz — CHANGES_REQUESTED** [review]

**Comment by cpu** [comment]
> I guess we'd need to write an encoder to put these strings into the form of a NameConstraints extension body, but that is definitely doable?

That's what I was thinking but wasn't sure if I was overlooking some inherent ambiguity that would make it harder than it seems.

**Comment by djc** [comment]
> Way back I had a test for this root's constraints -- unfortunately I never landed it because the test failed, because the webpki crate didn't have working name constraints support back then. That is fixed since January. The old commit is here: [65282d9](https://github.com/rustls/webpki-roots/commit/65282d9f07f46d9f272c82034811f9155920844b) - unfortunately only a positive test.

Oh great -- that was pretty easy to rebase onto the current state of affairs, and allowed me to verify that your suggested fix works.

**Review by ctz — APPROVED** [review]

**Review by cpu — APPROVED** [review]

**Comment by djc** [comment]
Published 0.25.1.

**Pull request merged**